mlflow/deployments/databricks/__init__.py

```diff
@@ -119,7 +206,46 @@ def create_endpoint(self, name, config=None):
     @experimental
     def update_endpoint(self, endpoint, config=None):
         """
-        TODO
+        Update a specified serving endpoint with the provided configuration.
+        See https://docs.databricks.com/api/workspace/servingendpoints/updateconfig for
+        request/response schema.
+
+        :param endpoint: The name of the serving endpoint to update.
+        :param config: A dictionary containing the configuration of the serving endpoint to update.
+        :return: A :py:class:`DatabricksEndpoint` object containing the request response.
+
+        Example:
+
+        .. code-block:: python
+
+            from mlflow.deployments import get_deploy_client
+
+            client = get_deploy_client("databricks")
+            client.update_endpoint(
+                endpoint="chat",
+                config={
+                    "served_entities": [
+                        {
+                            "name": "test",
+                            "external_model": {
+                                "name": "gpt-4",
+                                "provider": "openai",
+                                "task": "llm/v1/chat",
+                                "openai_config": {
+                                    "openai_api_key": "{{secrets/scope/key}}",
+                                },
+                            },
+                        }
+                    ],
+                    "rate_limits": [
+                        {
+                            "key": "user",
+                            "renewal_period": "minute",
+                            "calls": 10,
+                        }
+                    ],
+                },
+            )
```

Comment:
@harupy I thought @mingyu89 said that we can update limits without having to specify `served_entities`. Is that not accurate / did I misunderstand? 

If you *can* update limits without having to specify `served_entities`, can we make sure the implementation supports that and demonstrate how to do this in the example  code snippet above?

Reply:
To do that, we need to use `/api/2.0/serving-endpoints/{name}/rate-limits` endpoint.

Reply:
Yeah, we should do that. Else it's inconvenient for users to set limits. I thought we had that at some point in the previous PR(s). Can we add it back?

Reply:
I think the check should be simple: if `config` only has `rate_limits`, call the `/api/2.0/serving-endpoints/{name}/rate-limits` endpoint

Reply:
Our focus should be making this easier for the user, even if the implementation requires an extra if / else

Reply:
Sounds good, will file a PR.

Reply:
Thanks :)

Reply:
Filed https://github.com/mlflow/mlflow/pull/10502